cmd/yawol-cloud-controller/main.go

```diff
@@ -138,10 +139,12 @@ func main() {
 	}
 
 	controlMgr, err := ctrl.NewManager(getConfigFromKubeconfigOrDie(controlKubeconfig), ctrl.Options{
-		Scheme:                        scheme,
-		MetricsBindAddress:            "0",
-		Port:                          9443,
-		Namespace:                     *infrastructureDefaults.Namespace,
+		Scheme:             scheme,
+		MetricsBindAddress: "0",
+		NewCache: func(config *rest.Config, opts cache.Options) (cache.Cache, error) {
+			opts.Namespaces = append(opts.Namespaces, *infrastructureDefaults.Namespace)
+			return cache.New(config, opts)
+		},
```

Comment:
`NewCache` is a low-level primitive but not required here. You can use the cache options:
```suggestion
		Cache: cache.Options{
			Namespaces: []string{*infrastructureDefaults.Namespace},
		},
```